### PR TITLE
24004202 Add save to the db - Endpoint SearchBooksByTitle

### PR DESCRIPTION
**Pull request by tinkabel85**

### CODE REVIEW

**Comment by tinkabel85** [comment]
I reverted the path in routes.yaml back to  path: /books/{title} just to see if it is working. Otherwise I couldn't check

File: src/Controller/BookController.php [hunk 1]
```diff
@@ -29,13 +36,79 @@ public function index(): Response
 
   public function searchBooksByTitle(string $title): Response
   {
+    $title = urldecode($title);
```
Comment (anna-bc) [inline src/Controller/BookController.php:39]: remove this and look at next comment

File: src/Controller/BookController.php [hunk 2]
```diff
@@ -29,13 +36,79 @@ public function index(): Response
...
+    $bookRepository = $this->entityManager->getRepository(Book::class);
+    // $book = $bookRepository->findOneBy(['title' => $title]);
+    $books = $bookRepository->createQueryBuilder('b')
+    ->where('b.title LIKE :title')
+    ->setParameter('title', '%' . $title . '%')
```
Comment (anna-bc) [inline src/Controller/BookController.php:44]: add the urldecode($title) here:
`->setParameter('title', '%' . $title . '%')`

the urldecode($title) will work for our database, but not for the api, if the title contains spaces. 
With my suggested change both will work

**Review by anna-bc — CHANGES_REQUESTED** [review]

**Review by naomiinanna — DISMISSED** [review]

File: src/Controller/BookController.php (hunk 2 above)
Comment (tinkabel85) [reply src/Controller/BookController.php]: done!

**Review by naomiinanna — APPROVED** [review]

**Pull request merged**